packages/components/src/components/date-picker/date-picker-native.tsx

```diff
@@ -74,7 +92,11 @@ const Native = ({
                 {label || (!value && placeholder)}
             </label>
 
-            <Icon icon='IcCalendar' className='dc-datepicker__calendar-icon' color={disabled && 'disabled'} />
+            <Icon
+                icon='IcCalendar'
+                className='dc-datepicker__calendar-icon'
+                color={disabled ? 'disabled' : undefined}
```

Comment:
```suggestion
                color={disabled && 'disabled'}
```

Reply:
`color` prop on `Icon` component is expecting a `string` or `undefined` type, by doing `disabled && 'disabled'` typescript will complain because this way we are passing a boolean value to `color` prop which only expects `string` or `undefined`